### PR TITLE
Subscriptions: Add action execution and rendering actions

### PR DESCRIPTION
**Pull request by prateekbh**
- Adds event listener for actions
- Adds rendering logic for platforms
- Adds methods for Subscription platform Interface

### CODE REVIEW

File: extensions/amp-subscriptions/0.1/local-subscription-platform-renderer.js [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+  /**
+   * @param {!../../../src/service/ampdoc-impl.AmpDoc} ampdoc
+   */
+  constructor(ampdoc) {
+    /** @private @const */
+    this.ampdoc_ = ampdoc;
+
+    /** @private @const */
+    this.document_ = this.ampdoc_.win.document;
```
Comment (dvoytenko) [inline extensions/amp-subscriptions/0.1/local-subscription-platform-renderer.js]: You should almost never use document. Instead, there's `ampdoc.getRootNode()` and family.

File: extensions/amp-subscriptions/0.1/local-subscription-platform-renderer.js [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    return this.ampdoc_.whenReady().then(() => {
+      // Find the first matching dialog.
+      const actionCandidates =
+          this.document_.querySelectorAll('[subscriptions-action]');
+      for (let i = 0; i < actionCandidates.length; i++) {
+        const candidate = actionCandidates[i];
+        const expr = candidate.getAttribute('subscriptions-display');
+        if (expr && evaluateExpr(expr, entitlements)) {
+          candidate.setAttribute('i-amphtml-subs-display', '');
+        }
```
Comment (dvoytenko) [inline extensions/amp-subscriptions/0.1/local-subscription-platform-renderer.js]: If you do `else {candidate.removeAttribute(...)}` this method will become reentrant.

File: extensions/amp-subscriptions/0.1/local-subscription-platform.js [hunk 3]
```diff
@@ -110,6 +130,40 @@ export class LocalSubscriptionPlatform {
     return this.readerIdPromise_;
   }
 
+  /**
+   * Add event listener for the subscriptions action
+   * @private
+   */
+  initializeListeners_() {
+    this.document_.addEventListener('click', e => {
```
Comment (dvoytenko) [inline extensions/amp-subscriptions/0.1/local-subscription-platform.js]: Ditto. Only add listeners to `ampdoc.getRootNode()`

File: extensions/amp-subscriptions/0.1/local-subscription-platform.js [hunk 4]
```diff
@@ -110,6 +130,40 @@ export class LocalSubscriptionPlatform {
...
+
+  /**
+   * Executes action for the local platform.
+   * @param {string} action
+   */
+  executeAction(action) {
+    const actionExecution = this.actions_.execute(action);
+    actionExecution.then(result => {
+      if (result) {
+        this.getEntitlements();
```
Comment (dvoytenko) [inline extensions/amp-subscriptions/0.1/local-subscription-platform.js]: Would this have any affect on rendering or such? Also, you may want to execute this call via the top-level service b/c changing of entitlements here could result in selecting a new platform... Maybe something like: `service.reAuthorizePlatform(this)` or such?

File: extensions/amp-subscriptions/0.1/subscription-platform.js [hunk 5]
```diff
@@ -26,12 +26,25 @@ import {PageConfig} from '../../../third_party/subscriptions-project/config';
...
    * @return {!Promise<!Entitlements>}
    */
   getEntitlements() {
   }
+
+  /**
+   * Renders the UI specific to the platform
+   */
+  render() {
+
```
Comment (dvoytenko) [inline extensions/amp-subscriptions/0.1/subscription-platform.js]: remove extra line

File: extensions/amp-subscriptions/0.1/subscription-platform.js [hunk 6]
```diff
@@ -26,12 +26,25 @@ import {PageConfig} from '../../../third_party/subscriptions-project/config';
...
    * Requests entitlement for a subscription platform.
    * @return {!Promise<!Entitlements>}
    */
   getEntitlements() {
   }
+
+  /**
+   * Renders the UI specific to the platform
+   */
+  render() {
```
Comment (dvoytenko) [inline extensions/amp-subscriptions/0.1/subscription-platform.js]: Implement right away on google subscriptions: even if it's just an empty method. But I feel like the name is not broad enough. Maybe `activate()`?

File: extensions/amp-subscriptions/0.1/local-subscription-platform-renderer.js (hunk 1 above)
Comment (prateekbh) [reply extensions/amp-subscriptions/0.1/local-subscription-platform-renderer.js]: Done

File: extensions/amp-subscriptions/0.1/local-subscription-platform-renderer.js (hunk 2 above)
Comment (prateekbh) [reply extensions/amp-subscriptions/0.1/local-subscription-platform-renderer.js]: I believe that this should be a part of a separate `deactivate` method which would be called for all platforms which are not selected. WDYT?

File: extensions/amp-subscriptions/0.1/local-subscription-platform.js (hunk 3 above)
Comment (prateekbh) [reply extensions/amp-subscriptions/0.1/local-subscription-platform.js]: Done

File: extensions/amp-subscriptions/0.1/subscription-platform.js (hunk 6 above)
Comment (prateekbh) [reply extensions/amp-subscriptions/0.1/subscription-platform.js]: Done

File: extensions/amp-subscriptions/0.1/subscription-platform.js (hunk 5 above)
Comment (prateekbh) [reply extensions/amp-subscriptions/0.1/subscription-platform.js]: Done

File: extensions/amp-subscriptions/0.1/local-subscription-platform.js (hunk 4 above)
Comment (prateekbh) [reply extensions/amp-subscriptions/0.1/local-subscription-platform.js]: yes, added a todo which will resolve with the actio delegation PR as both need a service interface

File: extensions/amp-subscriptions/0.1/local-subscription-platform-renderer.js (hunk 2 above)
Comment (dvoytenko) [reply extensions/amp-subscriptions/0.1/local-subscription-platform-renderer.js]: Probably a good idea. Feel free to proceed with that idea in a separate PR.

**Pull request merged**